### ISSUE TITLE
Intermittent HTTP 409 failure when attempting to remove a custom hostname binding from an App Service deployment slot

### ISSUE THREAD

**Issue by ktenedios-nimbus** [hashicorp/terraform-provider-azurerm#27477]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.9.4

### AzureRM Provider Version

4.3.0

### Affected Resource(s)/Data Source(s)

azurerm_app_service_slot_custom_hostname_binding

### Terraform Configuration Files

```hcl
resource "azurerm_app_service_slot_custom_hostname_binding" "_signalrload-nimbus-cloud" {
  app_service_slot_id = "/subscriptions/1e1958d2-f3d9-4b66-a2cb-7677902d7cf6/resourceGroups/internal-test-ausespoke-rg/providers/Microsoft.Web/sites/internal-test-ausespoke-app-service/slots/previous-version"
  hostname            = "signalrload.nimbus.cloud"
  ssl_state           = "SniEnabled"
  thumbprint          = "CDC0458120AB88ABF648E1F009BFB9BFCB391735"


  depends_on = [

    azurerm_dns_a_record.application-gateway-a-record,
    azurerm_dns_txt_record.custom-domain-verification-id-txt-record
  ]

}
```

### Debug Output/Panic Output

```shell
https://gist.github.com/ktenedios-nimbus/f0140c23ccd32143e46058128d843d0a
```

### Expected Behaviour

Custom hostname binding in a deployment slot should have been deleted without any issues

### Actual Behaviour

The following error was thrown:
```
Error: deleting App Service Slot Custom Hostname Binding: (Host Name Binding Name "terraformdev02.nimbus.cloud" / Slot Name "terraform" / Site Name "dev-ausespoke1-app-service" / Resource Group "dev-ausespoke1-rg"): web.AppsClient#DeleteHostNameBindingSlot: Failure responding to request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=409 Code="Conflict" Message="There was a concurrency exception. Please try again." Details=[{"Message":"There was a concurrency exception. Please try again."},{"Code":"Conflict"},{"ErrorEntity":{"Code":"Conflict","ExtendedCode":"59406","Message":"There was a concurrency exception. Please try again.","MessageTemplate":"There was a concurrency exception. Please try again.","Parameters":[]}}]
```

Despite the error being thrown, the custom hostname binding was removed.

### Steps to Reproduce

terraform apply

**Comment by bdorplatt** [comment]
We are seeing the same error when destroying hostname bindings app service and slots

Error: deleting App Service Slot Custom Hostname Binding: web.AppsClient#DeleteHostNameBindingSlot: Failure responding to request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=409 Code="Conflict" Message="There was a concurrency exception. Please try again." Details=[{"Message":"There was a concurrency exception. Please try again."},{"Code":"Conflict"},{"ErrorEntity":{"Code":"Conflict","ExtendedCode":"59406","Message":"There was a concurrency exception. Please try again.","MessageTemplate":"There was a concurrency exception. Please try again.","Parameters":[]}}]
│

**Comment by brentlyjdavid** [comment]
i don't think this is limited to slot deployments. we have the same thing without using slots and are getting the 409 exceptions, it runs successful on a second destroy command.  not sure when it started, but feels like it is recent.  we are using managed certs, not a custom thumbprint as OP syntax suggests.

using provider version 4.3.0 and terraform version 1.9.7

`web.AppsClient#DeleteHostNameBinding: Failure responding to request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=409 Code="Conflict" Message="There was a concurrency exception. Please try again." Details=[{"Message":"There was a concurrency exception. Please try again."},{"Code":"Conflict"},{"ErrorEntity":{"Code":"Conflict","ExtendedCode":"59406","Message":"There was a concurrency exception. Please try again.","MessageTemplate":"There was a concurrency exception. Please try again.","Parameters":[]}}][0m`

**Comment by bdorplatt** [comment]
@apparentlymart @jackofallops 
After some additional testing, it appears this issue is affecting any destroy attempts across all AzureRM provider and Terraform versions. If an app has a custom hostname binding using either azurerm_app_service_slot_custom_hostname_binding or azurerm_app_service_custom_hostname_binding, the destroy will fail with the error below:

Error: deleting App Service Slot Custom Hostname Binding: web.AppsClient#DeleteHostNameBindingSlot: Failure responding to request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=409 Code="Conflict" Message="There was a concurrency exception. Please try again." Details=[{"Message":"There was a concurrency exception. Please try again."},{"Code":"Conflict"},{"ErrorEntity":
│ 

It isn't possible to even destroy a single hostname binding with nothing else being touched in the destroy. The concurrency appears to be with itself somehow.
